shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCX.scala

```diff
@@ -399,14 +399,14 @@ class UCX(executor: BlockManagerId, rapidsConf: RapidsConf) extends AutoCloseabl
     val result = ByteBuffer.allocateDirect(4 + hostnameBytes.length)
     result.putInt(listenerAddress.getPort)
     result.put(hostnameBytes)
+    result.rewind()
     result
   } else {
     worker.getAddress
   }
 
   private def getUcxAddress: ByteBuffer = {
     val result = ucxAddress.asReadOnlyBuffer()
```

Comment:
nit, this cold just be now `ucxAddress.asReadOnlyBuffer()`, no need for `val result`.